cli/testflinger_cli/__init__.py

```diff
@@ -425,6 +441,18 @@ def agent_status(self):
 
     def queue_status(self):
         """Show the status of the agents in a specified queue."""
+        if self.args.jobs:
+            # New job-focused functionality
+            self._queue_status_jobs()
+        else:
+            # Original agent-focused functionality
+            self._queue_status_agents()
```

Comment:
I like the idea you are proposing for separating the logic for getting status for agents/jobs! As a suggestion, I think we can make this `queue_status()` to do the output formatting and just use the helper functions to retrieve the agent/jobs status. 

e.g
```suggestion
    def queue_status(self):
        """Show agent and job status in a specified queue."""
            agents_status =  self._queue_status_agents() # This always returns all agents in queue and its status
            jobs_status = self._queue_status_jobs() # This returns "waiting" jobs by default or all jobs if verbose
            
            if self.args.json:
                output = json.dumps() # JSON parsing logic
            else:
                output = ("") # Human Readable Logic
             print(output) 
           
```